md-gravity/Icebreaker
# Add ability to send messages via socket

# Description

The user is now able to send messages to a specific room and receive messages from other users in that same room. However, at this point, the feature does not include a user interface.

## What Changed


The latest update includes a new implementation logic where the user creates a message in the `archivist `service, where all necessary checks are performed, before sending the message to the socket. Subsequently, the user subscribed to the room via `socket `receives this message.

It's important to note that the `archivist `service doesn't emit a message created event due to `NATS `being a bottleneck in this scenario. As a result, the `telegraph `could potentially have to wait too long to receive the message from `NATS`.

```
UI > Archivist Create Message > Message > UI
Message > UI > Telegraph Message
Telegraph On Message > UI
```

## How To Test

1. Create two users in the system, one as the sender and the other as the receiver.
2. Create a new room and make sure both users are added as members.
3. Open the created room with both users logged in to their respective accounts.
4. Use the message input form to send messages between the two users in the room.
5. Open the browser console to view the messages exchanged between the users.

# Checklist
- [x] I have performed a self-review(clean, lint, typecheck, build) of my code & follow full user flow
- [x] Code Improvement & Refactoring
